.github/workflows/staging-deployment.yml

```
@@ -0,0 +1,179 @@
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set build workflow ID
+        id: set_build_workflow_id
+        run: |
+          WORKFLOW_ID=$(echo '${{ steps.get_workflow_run.outputs.data }}' | jq -r '.workflow_runs[0].id')
+          echo "workflow_id=$WORKFLOW_ID" >> $GITHUB_OUTPUT
+  
+      - name: Check for war artifact
+        id: verify_artifact
+        uses: octokit/request-action@v2.x
+        with:
+          route: GET /repos/${{ github.repository }}/actions/runs/${{ steps.set_build_workflow_id.outputs.workflow_id }}/artifacts?name=Artemis.war
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  
+      - name: Verify artifact exists
+        id: check_result
+        run: |
+          TOTAL_COUNT=$(echo '${{ steps.verify_artifact.outputs.data }}' | jq -r '.total_count')
+            
+          if [ "$TOTAL_COUNT" -gt 0 ]; then
+            echo "Found Artemis.war artifact in latest build"
+          else
+            echo "::error::No Artemis.war artifact found in latest build!"
+            exit 1
+          fi
+
+  deploy:
+    needs: check-build-status
+    runs-on: github-l-4
+    environment: 
+        name: artemis-staging-localci.artemis.cit.tum.de
+        url: ${{ vars.DEPLOYMENT_URL }}
+    env:
+      DEPLOYMENT_HOSTS_PRIMARY: ${{ vars.DEPLOYMENT_HOSTS_PRIMARY}}
+      DEPLOYMENT_HOSTS_SECONDARY: "artemis-staging-localci-node-2.artemis.cit.tum.de artemis-staging-localci-node-3.artemis.cit.tum.de artemis-staging-localci-node-4.artemis.cit.tum.de artemis-staging-localci-node-5.artemis.cit.tum.de"
+      DEPLOYMENT_USER: ${{ vars.DEPLOYMENT_USER }}
+
+      DEPLOYMENT_FOLDER: ${{ vars.DEPLOYMENT_FOLDER }}
+
+      HEALTH_CHECK_URL: "${{vars.DEPLOYMENT_URL}}/management/health"
+      WORKFLOW_RUN_ID: ${{needs.check-build-status.outputs.build_workflow_run_id}}
+
+    timeout-minutes: 10
+    
+    steps:
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: Artemis.war
+          path: artifacts
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          run-id: ${{ env.WORKFLOW_RUN_ID  }}
+
+      - name: Setup SSH
+        env:
+          DEPLOYMENT_SSH_KEY: ${{ secrets.DEPLOYMENT_SSH_KEY }}
+          SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+          DEPLOYMENT_HOST_PUBLIC_KEYS: ${{ vars.DEPLOYMENT_HOST_PUBLIC_KEYS }}
+        run: |
+          mkdir -p ~/.ssh
+          ssh-agent -a $SSH_AUTH_SOCK > /dev/null
+          ssh-add - <<< "$DEPLOYMENT_SSH_KEY"
+          cat - <<< $DEPLOYMENT_HOST_PUBLIC_KEYS >> ~/.ssh/known_hosts
+          ssh-add -l || echo "No identities found"
+
+      - name: Phase 1 - Stop Secondary Nodes
+        run: |
+          for node in $DEPLOYMENT_HOSTS_SECONDARY; do
+            echo "[INFO] Stop artemis.service on ${node} ..."
+            ssh -o BatchMode=yes -o StrictHostKeyChecking=accept-new \
```

Do not do this. If the Host Key is wrong this is an issue that should not be ignored.
```suggestion
            ssh -o BatchMode=yes \
```